### PR TITLE
Remove SDK Architecture from 'docs' - Closes 1509

### LINKED ISSUE #1509: Update `SDK architecture`
### Description

- Move the contents of [SDK Architecture](https://lisk.com/documentation/understand-blockchain/architecture.html) to [Lisk SDK landing page](https://lisk.com/documentation/lisk-sdk/index.html)
- Delete the `architecture.adoc` file.
- Update navigation.

- Comment the following sections for now.
  - Application 
  - Genesis block
  - Configuration

### Affected Branch
 `docs-sdk`

### PR DESCRIPTION
**Pull request by TalhaMaliktz**
Closes #1509 
- Update URL reference and text in **rpc.adoc**
- Rename the **architecture.adoc** to **architecture-pre1509-old.adoc**
- Remove reference of **architecture.adoc** from files

### CODE REVIEW

File: docs/modules/ROOT/pages/understand-blockchain/sdk/rpc.adoc [hunk 1]
```diff
@@ -48,7 +49,8 @@ image::intro/communication-architecture.png[,1000 ,align="center"]
 [[the-api-client]]
 == The API client
 
-The xref:{url_references_elements_apiclient}[] simplifies sending API requests to a blockchain application. Blockchain applications developed with Lisk SDK communicate to a node via RPCs. RPC-based interfaces i.e. IPC, WS, or HTTP allow applications to *invoke* and *subscribe* (WS and IPC only)  to RPC endpoints and events. 
+The xref:{url_references_elements_apiclient}[] simplifies sending API requests to a blockchain application. 
+Blockchain applications developed with Lisk SDK communicate to a node via RPCs. RPC-based interfaces i.e. IPC, WS, or HTTP allow applications to *invoke* and *subscribe* (WS and IPC only)  to RPC endpoints and events.
```
Comment (chris529) [inline docs/modules/ROOT/pages/understand-blockchain/sdk/rpc.adoc]:
```suggestion
Blockchain applications developed with Lisk SDK communicate to a node via RPCs. RPC-based interfaces i.e. IPC, WS, or HTTP allow applications to *invoke* and *subscribe* (WS and IPC only) to RPC endpoints and events. 
```

**Review by chris529 — COMMENTED** [review]
Looks great! Only 1 very minor point 👍 )

**Review by chris529 — APPROVED** [review]
LGTM!

**Review by chris529 — APPROVED** [review]
LGTM! 👍

**Review by Tschakki — APPROVED** [review]
LGTM

**Pull request merged**